### PR TITLE
How to doc

### PR DESCRIPTION
**Pull request by cdquinto**
@agladstein and I wrote these paragraphs to explain how to add documentation. We are not really sure if the part where we described the structure of the documentation is accurate. Let us know if there is something we should change.
This should close https://github.com/tskit-dev/msprime/issues/1305 and probably https://github.com/tskit-dev/msprime/issues/1275. 
This is also related to https://github.com/tskit-dev/msprime/issues/1273.

### CODE REVIEW

File: docs/development.rst [hunk 1]
```diff
@@ -571,12 +571,48 @@ You can use ``docker`` to locally build an image, but it requires root access:
 Documentation
 *************
 
-Documentation is written using `Sphinx <http://www.sphinx-doc.org/en/stable/>`_
-and contained in the ``docs`` directory. It is written in the
-`reStructuredText <http://docutils.sourceforge.net/rst.html>`_ format and
-is deployed automatically to `readthedocs <https://readthedocs.org/>`_. To
-build the documentation locally run ``make`` in the ``docs`` directory.
-This should build the HTML documentation in ``docs/_build/html/``.
+To get started writing documentation for msprime, first follow the `Quickstart <https://tskit-dev.github.io/msprime-docs/main/development.html#quickstart>`_. Once you have created and checked out a "topic branch", you are ready to start editing the documentation.
```
Comment (jeromekelleher) [inline docs/development.rst:574]: Better to use sphinx cross references here, it'll be something like ":ref:`sec_development_quickstart`", but you might need to check the actual label is defined. This is more robust, especially as we're moving the docs into ``tskit.dev`` sometime soon.

File: docs/development.rst [hunk 2]
```diff
@@ -571,12 +571,48 @@ You can use ``docker`` to locally build an image, but it requires root access:
...
+
+The msprime documentation consists of three main sections - `Ancestry <https://tskit-dev.github.io/msprime-docs/main/ancestry.html>`_, `Mutations <https://tskit-dev.github.io/msprime-docs/main/mutations.html>`_, and `Demography <https://tskit-dev.github.io/msprime-docs/main/demography.html>`_. Each of these sections should include an introductory paragraph so users understand what that general section is used for. Within each of the main sections, there are subsections for all of the features. Each of the feature subsections should include a quick overview of what it is, definitions, and example usage. The end of each main section should also include the detailed API documentation.
```
Comment (jeromekelleher) [inline docs/development.rst:576]: Similarly here.

File: docs/development.rst [hunk 3]
```diff
@@ -571,12 +571,48 @@ You can use ``docker`` to locally build an image, but it requires root access:
...
+    .. jupyter-execute::
+
+        print('Python code!')
+
+
+The API documentation is automatically generated using `docstrings <https://www.python.org/dev/peps/pep-0257/>`_. Therefore, edits to the Python API documentation must be made within the source code.
+
+.. todo:: Describe what the API documentation should include. (e.g. Each function should include a small description of what it does, the input parameters, and returned parameters.)
+
+To build the documentation locally, go to the ``docs`` directory and run ``make``. This will build the HTML documentation in  ``docs/_build/html/``. You can now view the local build of the HTML in your local browser (if you do not know how to do this, try double clicking the HTML file). Building the documentation will also run any new jupyter-sphynix code, so it may take a while.
```
Comment (jeromekelleher) [inline docs/development.rst:613]: typo, "sphynix"

**Review by jeromekelleher — COMMENTED** [review]
This is great, thanks @cdquinto! Some minor comments.

**Comment by cdquinto** [comment]
I changed the references for the big sections and corrected the typo.

**Comment by jeromekelleher** [comment]
@mergifyio rebase

**Comment by jeromekelleher** [comment]
Excellent; thanks @cdquinto!

**Comment by jeromekelleher** [comment]
@Mergifyio rebase

**Pull request merged**